lib/solvers/AutoroutingPipelineSolver.ts

```diff
@@ -170,12 +174,34 @@ export class CapacityMeshSolver extends BaseSolver {
       "edgeSolver",
       CapacityMeshEdgeSolver2_NodeTreeOptimization,
       (cms) => [cms.capacityNodes!],
+      {
+        onSolved: (cms) => {
+          cms.capacityEdges = cms.edgeSolver?.edges!
+        },
+      },
+    ),
+    definePipelineStep(
+      "deadEndSolver",
+      DeadEndSolver,
+      (cms) => [{ nodes: cms.capacityNodes!, edges: cms.capacityEdges! }],
+      {
+        onSolved: (cms) => {
+          const removedNodeIds = cms.deadEndSolver?.removedNodeIds!
+
+          cms.capacityNodes = cms.capacityNodes!.filter(
+            (n) => !removedNodeIds.has(n.capacityMeshNodeId),
+          )
+          cms.capacityEdges = cms.capacityEdges!.filter((e) =>
+            e.nodeIds.every((nodeId) => !removedNodeIds.has(nodeId)),
+          )
```

Comment:
@seveibar as I understand this should be sufficient to remove the dead-ends from the state. But might be worth double checking that I did not missed anything else.